nym-vpn-core/nym-vpn-lib/src/lib.rs

```diff
@@ -187,13 +187,13 @@ pub struct NymVpn<T: Vpn> {
     shadow_handle: ShadowHandle,
 }
 
-#[derive(Debug)]
+#[derive(Debug, PartialEq, uniffi::Record, Clone)]
 pub struct MixnetConnectionInfo {
     pub nym_address: Recipient,
     pub entry_gateway: NodeIdentity,
 }
 
-#[derive(Debug)]
+#[derive(Debug, PartialEq, uniffi::Record, Clone)]
```

Comment:
Imho we should try to keep uniffi separate from this file. Ideally I think we should go in the other direction where the uniffi stuff is in a separate crate nym-vpn-uniffi

Reply:
Is this something we want to do now or later?

Reply:
No need to create a new crate in this PR, I was just thinking that it would be good to keep uniffi stuff separate and confined inside `platform` and `uniffi_custom_impls.rs` so that we don't make this split harder in the future

Reply:
I concur, I think we should keep uniffi isolated and convert to/from those types.

Reply:
So i changed this, but uniffi is having issues generating now. Any ideas?

nym_vpn_lib.kt:2317:17 Argument type mismatch: actual type is 'nym_vpn_lib.ConnectionInfo', but 'nym_vpn_lib.NymVpnStatus.ConnectionInfo' was expected.